cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -291,7 +292,6 @@ public synchronized ProgramController start(ProgramId programId, Map<String, Str
    */
   public ProgramRuntimeService.RuntimeInfo start(final ProgramId programId, final Map<String, String> systemArgs,
```

Comment:
It would be better to change the method name to indicate that this is an internal method, and should not be called from an external service. Also good to add this detail in the javadoc.